src/common/mfu_daos.c

```
@@ -1183,18 +1191,22 @@ int daos_connect(
 
         /* check container status, and if unhealthy do not continue unless --force
          * option is used (only for serialization, never for copies) */
-        rc = daos_check_cont_status(*coh);
-        if (rc != 0 && !force_serialize) {
+        bool status_healthy;
+        rc = daos_check_cont_status(*coh, &status_healthy);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Checking DAOS container status failed: %s", strerror(rc));
```

This log might print an unhelpful strerror, since daos_check_cont_status returns 1 on most errors

---

@daltonbohning would it be better to return the DAOS error code? Seems like a lot of the functions just return 1 on error, which yeah is not very helpful I guess.

---

@dsikich It's tricky, because it isn't always simple / straightforward to aggregate error codes across processes. And even then, we would need to adjust every call all the way up to daos-serialize.c to NOT just return 1.
In this case, maybe it makes sense for daos_check_cont_status itself to log a message with the corresponding error code, but still return 1. Then, this part of the code can just return without printing anything, or at least without printing an error code.

---

Ultimately the idea is that the user shouldn't be confused about what the problem is

---

@daltonbohning I updated this to not print an error code, and daos_check_cont_status prints the DAOS error that occurred but the function returns 1.